### PR TITLE
datasets module: save\load fr_adience dataset to\from hdf5 file

### PR DESCRIPTION
**Pull request by terfendail**
### Recreated #139 by @avdmitry:
Code rebased to latest master to handle merge conflicts
### Original PR description
save\load fr_adience dataset to\from hdf5 file

### CODE REVIEW

File: modules/datasets/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,25 @@
 set(the_description "datasets framework")
+
+# find HDF5 library
+find_package(HDF5)
+if (HDF5_FOUND)
+    include_directories(${HDF5_INCLUDE_DIRS})
+    add_definitions(-DHAVE_HDF5=1)
```
Comment (alalek) [inline modules/datasets/CMakeLists.txt]: Don't change compiler flags before `ocv_define_module()`

File: modules/datasets/include/opencv2/datasets/util_hdf5.hpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+#ifdef HAVE_HDF5
+#ifndef OPENCV_DATASETS_UTIL_HDF5_HPP
+#define OPENCV_DATASETS_UTIL_HDF5_HPP
+
+#include <string>
+#include <vector>
+
+#include <opencv2/core.hpp>
+
+#include <hdf5.h>
```
Comment (alalek) [inline modules/datasets/include/opencv2/datasets/util_hdf5.hpp]: 3rdparty includes from OpenCV includes are not allowed.

File: modules/datasets/CMakeLists.txt [hunk 3]
```diff
@@ -1,4 +1,25 @@
...
+    message(STATUS "datasets: HDF5:   NO")
+endif()
+
 ocv_define_module(datasets opencv_core opencv_imgcodecs opencv_ml opencv_flann opencv_text WRAP python)
 
 ocv_warnings_disable(CMAKE_CXX_FLAGS /wd4267) # flann, Win64
+
+ocv_target_link_libraries(opencv_datasets LINK_PRIVATE ${HDF5_LIBRARIES})
+if(BUILD_EXAMPLES)
+    ocv_target_link_libraries(example_datasets_fr_adience_toh5 LINK_PRIVATE ${HDF5_LIBRARIES})
```
Comment (alalek) [inline modules/datasets/CMakeLists.txt]: This never works. Did you build your patch?

**Comment by terfendail** [comment]
Close PR, since there is separate module that provide HDF format parsing

**Pull request closed without merging**